src/internals/tuples/Tuples.ts

```diff
@@ -544,24 +589,60 @@ export namespace Tuples {
     return: PartitionImpl<Fn.arg0<this, Fn>, Fn.arg1<this, any[]>>;
   }
 
+  type ZipWithImpl<
+    arr1 extends unknown[],
+    arr2 extends unknown[],
+    fn extends Fn,
+    acc extends unknown[] = []
+  > = [arr1, arr2] extends [
+    [infer item1, ...infer rest1],
+    [infer item2, ...infer rest2]
+  ]
+    ? ZipWithImpl<rest1, rest2, fn, [...acc, Call<fn, [item1, item2]>]>
```

Comment:
I feel like the api would make more sense if the function was receiving items as two separate arguments instead of a tuple. I think this is how lodash's zipWith works

Reply:
Done, added a `Tuples.Create` since `F.Identity` would then not work anymore
There was a bug with the current implementation limiting Apply to only 5 parameters that used `RemoveUnknownArrayConstraint`
i removed it everywhere and the bug described that was supposed to be fixed by did not appear.
So maybe i was missing some use cases that where not covered by tests ?
I saw a failing test :  
```ts
it("should support intersections with unknown[] coming from function application", () => {
      type pipedArgs2 = unknown[] & [0, 1];
      type partialArgs2 = [unset, unset];
      type res2 = MergeArgs<pipedArgs2, partialArgs2>;
      type test2 = Expect<Equal<res2, [0, 1]>>;
    });
```
i removed it, since i don't see any use case that showed this in the code elsewhere.